packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -642,6 +642,28 @@ export class ConnectionService {
     return connectionRecord
   }
 
+  public async addConnectionType(agentContext: AgentContext, connectionRecord: ConnectionRecord, type: string) {
+    const tags = (connectionRecord.getTag('connectionType') as string[]) || ([] as string[])
+    connectionRecord.setTag('connectionType', [type, ...tags])
+    await this.update(agentContext, connectionRecord)
+  }
+
+  public async removeConnectionType(agentContext: AgentContext, connectionRecord: ConnectionRecord, type: string) {
+    const tags = (connectionRecord.getTag('connectionType') as string[]) || ([] as string[])
+
+    const newTags = tags.filter((value: string) => {
+      if (value != type) return value
+    })
+    connectionRecord.setTag('connectionType', [...newTags])
+
+    await this.update(agentContext, connectionRecord)
+  }
+
+  public async getConnectionTypes(connectionRecord: ConnectionRecord) {
+    const tags = connectionRecord.getTag('connectionType') as string[]
```

Comment:
This cast is incorrect (as it could be null), but per my comment above, not sure if the cast is needed anymore